pytest_jira.py

```diff
@@ -157,30 +163,37 @@ def __init__(
 
         # Setup basic_auth
         if self.username and self.password:
-            basic_auth = (self.username, self.password)
+            self.basic_auth = (self.username, self.password)
         else:
-            basic_auth = None
-
-        # TODO - use requests REST API instead to drop a dependency
-        # (https://confluence.atlassian.com/display/DOCSPRINT/
-        # The+Simplest+Possible+JIRA+REST+Examples)
-        self.jira = JIRA(
-            options=dict(server=self.url, verify=self.verify),
-            basic_auth=basic_auth,
-            validate=bool(basic_auth),
-            max_retries=1
-        )
+            self.basic_auth = None
+
+    def _jira_request(self, url, method='get', **kwargs):
+        if self.basic_auth:
+            return requests.request(method, url, auth=self.basic_auth, **kwargs)
+        else:
+            return requests.request(method, url, **kwargs)
+
+    def check_connection(self):
+        auth_url = '{url}/rest/auth/1/session'.format(url=self.url)
+        try:
+            r = self._jira_request(auth_url)
+            r.raise_for_status()
+            return True
+        except RequestException:
+            return False
```

Comment:
The reason why tests are failing that your solution doesn't support anonymous access to jira.
Put here 'raise' keyword and it will raise exception with auth error.